src/gwt/src/org/rstudio/core/client/widget/find/FindBar.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * FindBar.java
+ *
+ * Copyright (C) 2025 by Posit Software, PBC
+ *
+ * Unless you have received this program directly from Posit Software pursuant
+ * to the terms of a commercial license agreement with Posit Software, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.core.client.widget.find;
+
+import org.rstudio.core.client.CoreClientConstants;
+import org.rstudio.core.client.ElementIds;
+import org.rstudio.core.client.resources.ImageResource2x;
+import org.rstudio.core.client.theme.res.ThemeResources;
+import org.rstudio.core.client.theme.res.ThemeStyles;
+import org.rstudio.core.client.widget.DecorativeImage;
+import org.rstudio.core.client.widget.FindTextBox;
+import org.rstudio.core.client.widget.SmallButton;
+import org.rstudio.studio.client.workbench.views.history.view.Shelf;
+
+import com.google.gwt.aria.client.Roles;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.resources.client.ClientBundle;
+import com.google.gwt.resources.client.CssResource;
+import com.google.gwt.user.client.ui.Button;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.HasVerticalAlignment;
+import com.google.gwt.user.client.ui.HorizontalPanel;
+import com.google.gwt.user.client.ui.VerticalPanel;
+
+import elemental2.dom.Element;
+import elemental2.dom.KeyboardEvent;
+import jsinterop.base.Js;
+
+public abstract class FindBar extends Composite
+{
+   public enum Direction { NEXT, PREV };
+
+   public abstract void show(boolean focus);
+   public abstract void find(Direction dir);
+   public abstract void hide();
+
+   interface Resources extends ClientBundle
+   {
+      @Source("FindBar.css")
+      Styles styles();
+   }
+
+   interface Styles extends CssResource
+   {
+      String findBar();
+      String findTextBox();
+      String findPanel();
+      String closeButton();
+   }
+
+   public FindBar()
```

Comment:
This is a trimmed-down version of the existing FindReplaceBar, but decoupled from the commands system.